rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/ai/TbAiNodeConfiguration.java

```diff
@@ -41,6 +43,8 @@ public class TbAiNodeConfiguration implements NodeConfiguration<TbAiNodeConfigur
     @Length(min = 1, max = 500_000)
     private String userPrompt;
 
+    private List<UUID> resourceIds;
```

Comment:
```suggestion
    private Set<@NotNull(message = "references to resources cannot be null") TbResourceId> resources;
```

This is a minor suggestion, but I think it could be worth considering for a few reasons:

1. Ensure uniqueness by using `Set`
It doesn't really make sense to have duplicate resource IDs in the rule node configuration, so changing from `List` to `Set` would better express this constraint. This might also help in edge cases where users import corrupted rule chains.
2. Ensure that elements in the set cannot be null
Similar reasoning as above - null elements don't make sense in this context, so it would be good to disallow them explicitly.
3, Consistency
The `modelId` field is already present and uses the full ID object (with entity type). It would be nice to stay consistent and use the same format for resource IDs as well.

As a bonus, after this change you could remove the UUID-to-ID object mapping in the init method.